### PR TITLE
bug/renamed-columns-datespine

### PR DESCRIPTION
**Pull request by fivetran-catfritz**
## PR Overview
**This PR will address the following Issue/Feature:**
- T-707421

**This PR will result in the following new package version:**

- v1.1.1 since this is a logic change and is a patch to the last release.

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

> ## Bugfix
> - Updated the logic for model `int_salesforce__date_spine` to reference the `stg_*` staging models instead of the source tables.
>   - This was necessary since the staging models account for multiple spellings of column names while the source tables do not.

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt run –full-refresh && dbt test
- [ ] ~dbt run (if incremental models are present) && dbt test~

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked, tagged, and properly assigned
- [x] All necessary documentation and version upgrades have been applied
    
- [x] docs were regenerated (unless this PR does not include any code or yml updates)
- [x] BuildKite integration tests are passing
- [x] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:

#### Making sure the issue is corrected
1. Created seed table with column `CreatedDate` and without `created_date`.
2. Reproduced the issue with the production package
  - Screenshot 2024-07-23 at 3 45 08 PM
3. Corrected the issue with this update
  - Screenshot 2024-07-23 at 3 58 14 PM

#### Ensuring consistency between versions
- Final model `salesforce__daily_activity` is downstream of the date spine.
- Screenshot 2024-07-23 at 3 36 57 PM

### If you had to summarize this PR in an emoji, which would it be?
 
🎏

### CODE REVIEW

File: models/salesforce/intermediate/int_salesforce__date_spine.sql [hunk 1]
```diff
@@ -1,6 +1,11 @@
+{% if var('salesforce__lead_enabled', True) -%}
+-- depends_on: {{ var('lead') }}
+{% else -%}
+-- depends_on: {{ var('opportunity') }}
+{% endif %}
```
Comment (fivetran-catfritz) [inline models/salesforce/intermediate/int_salesforce__date_spine.sql:5]: This is still necessary for `dbt run`. I tried running without it, and the error said to put it back. It's still not necessary for compilation, however.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
+# dbt_salesforce v1.1.1
+[PR #56](https://github.com/fivetran/dbt_salesforce/pull/56) includes the following updates:
+## Bugfix
+- Updated the logic for model `int_salesforce__date_spine` to reference the `stg_*` staging models instead of the source tables.
+  - This was necessary since the staging models account for multiple spellings of column names while the source tables do not.
```
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:5]: Can you also add either another bullet here or an under the hood entry that now that model has a depends on statement and also leverages the flags.WHICH. More for our purposes so we know how this issue was addressed and some underlying impacts.

**Review by fivetran-joemarkiewicz — APPROVED** [review]
@fivetran-catfritz thanks for the quick resolution here! This change looks good to me. I just have one request to add an additional note to the CHANGELOG; however, that doesn't need to be blocking the review - as long as it is applied before the release review is initiated. Thanks!

File: CHANGELOG.md (hunk 2 above)
Comment (fivetran-catfritz) [reply CHANGELOG.md:5]: Updated!

**Pull request merged**